### PR TITLE
Calculate an execution cost scalar

### PR DESCRIPTION
**Pull request by kantai**
This implements #1886, by providing a method to collapse an execution cost into a scalar value, given a block limit. It does this by calculating the dot product of the following vectors:

```
<runtime, read_count, write_count, read_length, write_length> ·
 <1/runtime_limit, 1/read_count_limit, 1/write_count_limit, 1/read_length_limit, 1/write_length_limit>
```

By itself, this PR doesn't make a ton of progress on addressing the outstanding block fee calculation and rewards issues, but I wanted to open this to first get a sense of what would be required for wrapping up the fee calculation and rewards (relevant issues are: #1190 and #1695).

### CODE REVIEW

File: src/vm/costs/mod.rs [hunk 1]
```diff
@@ -302,6 +302,42 @@ impl ExecutionCost {
...
+                (self.write_length, limit.write_length),
+            ] {
+                let dimension_score =
+                    (*self_dimension as u128) * dimensional_max / (*lim_dimension as u128);
+                total_score += dimension_score;
+            }
+            // divide by the number of dimensions.
+            total_score /= 5;
+
+            Some(u64::try_from(total_score).expect("BUG: overflowed u64 calculating fill_scalar"))
```
Comment (jcnelson) [inline src/vm/costs/mod.rs:337]: This will _never_ fail, right?

Comment (kantai) [reply src/vm/costs/mod.rs:337]: That's right -- one of the unit tests supplies a maximum "stress" on this calculation.

**Review by jcnelson — APPROVED** [review]
LGTM, but as you point out, this may be moot if we're not going to implement #1190

**Review by lgalabru — APPROVED** [review]
Looks good to me!

**Comment by kantai** [comment]
Closing this PR, as this block fill evaluation is no longer required -- we're using a "first-price auction", meaning that in practice, tx fees are assessed according to the tx_fee on the transaction (rather than using block fill to calculate the second-price).

**Pull request closed without merging**